src/commands/lambda/uninstrument.ts

```diff
@@ -198,20 +199,54 @@ export class UninstrumentCommand extends Command {
 
     // Un-instrument functions.
     if (willUpdate) {
-      const promises = Object.values(configGroups).map((group) =>
-        updateLambdaFunctionConfigs(group.lambda, group.cloudWatchLogs, group.configs)
-      )
-      const spinner = renderer.updatingFunctionsSpinner(promises.length)
-      spinner.start()
-      try {
-        await Promise.all(promises)
-        spinner.succeed(renderer.renderUpdatedLambdaFunctions(promises.length))
-      } catch (err) {
-        this.context.stdout.write(renderer.renderFailureDuringUpdateError(err))
-        spinner.fail(renderer.renderFailedUpdatingLambdaFunctions())
+      const totalFunctions = Object.values(configGroups).reduce((c, group) => (c += group.configs.length), 0)
+      let totalFailedUpdates = 0
+      for (const group of configGroups) {
```

Comment:
The loop construct here and for the `instrument` method. Is it worthwhile to extract and combine this logic? Usually I wait until the 3rd time I need something to extract it, but I worry given the complexity here that someone may update this method or the `instrument` method and forget to mirror it here.

Reply:
You are right, previously it was mirrored but the logic seemed to be really simple, with the added complexity, it might cause an issue. Let me think how I can abstract it given the nature of the output writing.

Reply:
Solved, please let me know what you think of this new alternative.